hle-survey/transformation/src/main/scala/org/broadinstitute/monster/dap/dog/RoutineEnvironmentTransformations.scala

```diff
@@ -0,0 +1,370 @@
+package org.broadinstitute.monster.dap.dog
+
+import org.broadinstitute.monster.dap.RawRecord
+import org.broadinstitute.monster.dogaging.jadeschema.fragment.HlesDogRoutineEnvironment
+
+object RoutineEnvironmentTransformations {
+
+  /**
+    * Parse all routine environment fields out of a raw RedCap record,
+    * injecting them into a partially-modeled dog record.
+    */
+  def mapRoutineEnvironment(rawRecord: RawRecord): HlesDogRoutineEnvironment = {
+    val init = HlesDogRoutineEnvironment.init()
+
+    val transformations = List(
+      mapDogparks _,
+      mapRecreationalSpaces _,
+      mapWork _,
+      mapSitter _,
+      mapEatsFeces _,
+      mapToys _,
+      mapSleepLocation _,
+      mapToxinsIngested _,
+      mapOtherAnimals _,
+      mapGeneralRoutine _
+    )
+
+    transformations.foldLeft(init)((acc, f) => f(rawRecord, acc))
+  }
+
+  /* Dog Parks */
+  def mapDogparks(
+    rawRecord: RawRecord,
+    dog: HlesDogRoutineEnvironment
+  ): HlesDogRoutineEnvironment = {
+    val dogPark = rawRecord.getOptionalBoolean("de_dogpark_yn")
+    // ONLY if: de_dogpark_yn = 1
+    if (dogPark.contains(true)) {
+      val dogparkTravel = rawRecord.get("de_dogpark_get_to")
+      val dogparkTravelOther = dogparkTravel.map(_.contains("98"))
+      dog.copy(
+        deDogpark = dogPark,
+        deDogparkDaysPerMonth = rawRecord.getOptionalNumber("de_dogpark_freq"),
+        deDogparkTravelWalk = dogparkTravel.map(_.contains("1")),
+        deDogparkTravelDrive = dogparkTravel.map(_.contains("2")),
+        deDogparkTravelBike = dogparkTravel.map(_.contains("3")),
+        deDogparkTravelPublicTransportation = dogparkTravel.map(_.contains("4")),
+        deDogparkTravelOther = dogparkTravelOther,
+        //ONLY if: de_dogpark_get_to(98) = '1'
+        deDogparkTravelOtherDescription =
+          if (dogparkTravelOther.contains(true)) rawRecord.getOptional("de_dogpark_get_to_other")
+          else None,
```

Comment:
Style nitpick: Can you wrap the branches in curly brackets for the if/elses that spill to multiple lines?
```scala
if (thing) {
  // stuff
} else {
  // other stuff
}
```

Reply:
Same for all the conditionals below